### PR TITLE
enhancement(log):opt scheduled task log

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
The scheduled task logs are difficult to review. We are now adding a unique ID to locate task logs more easily.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-server/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -47,6 +47,20 @@
             </DefaultRolloverStrategy>
         </RollingFile>
 
+        <RollingFile name="SchedulerFileAppender" fileName="${LOG_DIRECTORY}/scheduler.log"
+                     filePattern="${LOG_DIRECTORY}/odc-%d{yyyy-MM-dd}-%i.log">
```
Comment (ungreat) [inline server/odc-server/src/main/resources/log4j2.xml:51]: filePattern should have the same prefix with filePattern.

File: server/odc-server/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -752,6 +766,14 @@
             <AppenderRef ref="ScheduleTaskWarnRoutingAppender"/>
         </Logger>
 
+        <Logger name="com.oceanbase.odc.service.quartz" level="INFO" additivity="true">
+            <AppenderRef ref="SchedulerFileAppender"/>
+        </Logger>
+
+        <Logger name="com.oceanbase.odc.service.schedule.job" level="INFO" additivity="true">
```
Comment (yhilmare) [inline server/odc-server/src/main/resources/log4j2.xml:773]: partitionplan job is using this appender 'AllRoutingPartitionPlanAppender'

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/OdcJobListener.java [hunk 3]
```diff
@@ -92,6 +93,10 @@ public void jobToBeExecuted(JobExecutionContext context) {
         ScheduleEntity scheduleEntity =
                 scheduleRepository.findById(scheduleId)
                         .orElseThrow(() -> new NotFoundException(ResourceType.ODC_SCHEDULE, "id", scheduleId));
+        ScheduleTaskContextHolder.trace(scheduleEntity.getId(), scheduleEntity.getJobType().name(), null);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/OdcJobListener.java:96]: why should we set these thread local? 'SchedulerFileAppender' seems like don't need any variables

**Review by yhilmare — APPROVED** [review]
lgtm, btw, we may test for partition plan job, cause that there  has been an appender for it~

**Pull request merged**